v1/plugins/logs/eventBuffer.go

```diff
@@ -0,0 +1,240 @@
+package logs
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/open-policy-agent/opa/v1/plugins/rest"
+)
+
+// eventBuffer stores and uploads a gzip compressed JSON array of EventV1 entries
+type eventBuffer struct {
+	buffer               chan []byte    // buffer stores JSON encoded EventV1 data
+	mtx                  sync.RWMutex   // mtx controls access to the buffer
+	payload              *payloadBuffer // payload contains the compressed JSON array to be uploaded
+	client               rest.Client    // client is used to upload the data to the configured service
+	uploadPath           string         // uploadPath is the configured HTTP resource path for upload
+	uploadSizeLimitBytes int64          // uploadSizeLimitBytes will enforce a maximum payload size to be uploaded
+}
+
+func newEventBuffer(bufferSizeLimitEvents int64, client rest.Client, uploadPath string, uploadSizeLimitBytes int64) *eventBuffer {
+	return &eventBuffer{
+		buffer:               make(chan []byte, bufferSizeLimitEvents),
+		payload:              newPayloadBuffer(),
+		client:               client,
+		uploadPath:           uploadPath,
+		uploadSizeLimitBytes: uploadSizeLimitBytes,
+	}
+}
+
+// Reconfigure updates the user configurable values
+func (b *eventBuffer) Reconfigure(bufferSizeLimitEvents int64, client rest.Client, uploadPath string, uploadSizeLimitBytes int64) {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+
+	b.client = client
+	b.uploadPath = uploadPath
+	b.uploadSizeLimitBytes = uploadSizeLimitBytes
+
+	if int64(cap(b.buffer)) == bufferSizeLimitEvents {
+		return
+	}
+
+	close(b.buffer)
+	newBuffer := make(chan []byte, bufferSizeLimitEvents)
+
+	for event := range b.buffer {
+		if int64(len(event)) < b.uploadSizeLimitBytes {
+			push(newBuffer, event)
+			continue
+		}
+
+		var decoded EventV1
+		if err := json.Unmarshal(event, &decoded); err != nil {
+			continue
+		}
+		if err := b.dropNDCache(&decoded, event); err != nil {
+			if !errors.As(err, &droppedNDCache{}) {
```

Comment:
[nit] Is it possible to use [`errors.Is`](https://pkg.go.dev/errors#Is) instead here? It seems applicable since we don't care for the actual error contents.

Reply:
Good catch, `errors.Is` is better to use. Changed.